egs/discophone/v1/local/chain/tuning/run_tdnn.sh

```diff
@@ -0,0 +1,275 @@
+#!/usr/bin/env bash
+
+# Siyuan Feng: based on babel/s5d/local/chain/tuning/run_tdnn.sh
+
+set -e -o pipefail
+
+# BABEL TRAIN:
+# Amharic - 307
+# Bengali - 103
+# Cantonese - 101
+# Javanese - 402
+# Vietnamese - 107
+# Zulu - 206
+# BABEL TEST:
+# Georgian - 404
+# Lao - 203
+babel_langs="307 103 101 402 107 206 404 203"
+babel_recog="${babel_langs}"
+data_aug_suffix= # set to null, whereas typically kaldi uses _sp
+
+#GlobalPhone:
+#Czech       S0196
+#French      S0197
+#Spanish     S0203
+#mandarin    S0193
+#Thai        S0321
+
+gp_langs="Czech French Mandarin Spanish Thai"
+gp_recog="${gp_langs}"
+
+
+
+# First the options that are passed through to run_ivector_common.sh
+# (some of which are also used in this script directly).
+stage=0
+stop_stage=1
+nj=30
+nj_align_fmllr_lats=1
+num_jobs_initial=1
+num_jobs_final=1
+#train_set=train
+gmm=tri5  # the gmm for the target data
+num_threads_ubm=12
+nnet3_affix=  # cleanup affix for nnet3 and chain dirs, e.g. _cleaned
+
+# The rest are configs specific to this script.  Most of the parameters
+# are just hardcoded at this level, in the commands below.
+train_stage=-10
+train_exit_stage=10000
+tree_affix=  # affix for tree directory, e.g. "a" or "b", in case we change the configuration.
+tdnn_affix=1a  #affix for TDNN directory, e.g. "a" or "b", in case we change the configuration.
+#common_egs_dir=  # you can set this to use previously dumped egs.
+chunk_width=150,120,90,75
+# End configuration section.
+echo "$0 $@"  # Print the command line for logging
+
+. ./cmd.sh
+. ./path.sh
+. ./utils/parse_options.sh
+
+train_set=""
+#train_set_hires=""
+dev_set=""
+#dev_set_hires=""
+for l in  ${babel_langs}; do
+  train_set="$l/data/train_${l} ${train_set}"
+#  train_set_hires="$l/data_mfcc_hires/train_${l} ${train_set_hires}"
+
+  dev_set="$l/data/dev_${l} ${dev_set}"
+#  dev_set_hires="$l/data_mfcc_hires/dev_${l} ${dev_set_hires}"
+done
+
+for l in ${gp_langs}; do
+  train_set="GlobalPhone/gp_${l}_train ${train_set}"
+  dev_set="GlobalPhone/gp_${l}_dev ${dev_set}"
+done
+
+
+train_set=${train_set%% }
+#train_set_hires=${train_set_hires%% }
+
+dev_set=${dev_set%% }
+#dev_set_hires=${dev_set_hires%% }
+
+recog_set=""
+for l in ${babel_recog} ${gp_recog}; do
+  recog_set="eval_${l} ${recog_set}"
+done
+recog_set=${recog_set%% }
+
+echo "Training data directories: ${train_set[*]}"
+echo "Dev data directories: ${dev_set[*]}"
+echo "Eval data directories: ${recog_set[*]}"
+
+full_train_set=train
+full_dev_set=dev
+
+function langname() {
+  # Utility
+  echo "$(basename "$1")"
+}
+
+
+if ! cuda-compiled; then
+  cat <<EOF && exit 1
+This script is intended to be used with GPUs but you have not compiled Kaldi with CUDA
+If you want to use GPUs (and have them), go to src/, and configure and make on a machine
+where "nvcc" is installed.
+EOF
+fi
+
+#local/chain/run_ivector_common.sh --stage $stage \
```

Comment:
why is ivector commented out?

Reply:
can we pass the `gp_langs`, `babel_langs`, and other language-set related variables to the ivector script? It will make it easier to run different experiment configurations going forward

Reply:
(same goes for the other run_tdnn script)

Reply:
> why is ivector commented out?

I forgot to un-comment local/chain/run_ivector_common.sh. I run it separately from running run_tdnn.sh. 
I am far from being familiar with Github and I realize code that I prepared earlier is not release-ready. I would need to clean up a little bit.

Reply:
> can we pass the `gp_langs`, `babel_langs`, and other language-set related variables to the ivector script? It will make it easier to run different experiment configurations going forward

That's what I did exactly. Since I run run_ivector_common.sh separatly I didn't update this inside run_tdnn.sh. I will update this.

Reply:
> (same goes for the other run_tdnn script)

Done